src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -769,9 +770,50 @@ HRESULT _stdcall TerminalResize(void* terminal, COORD dimensions)
     publicTerminal->_terminal->ClearSelection();
     publicTerminal->_renderer->TriggerRedrawAll();
 
+    // Resize the text buffer without changing the renderer draw size.
     return publicTerminal->_terminal->UserResize(dimensions);
 }
 
+/// <summary>
+/// Resizes the renderer to fit the provided width and height without changing the text
+/// buffer size.
+/// </summary>
+HRESULT _stdcall ResizeRendererDrawSpace(void* terminal, double width, double height, _Out_ COORD* dimensions)
+{
+    RETURN_HR_IF_NULL(E_INVALIDARG, dimensions);
+
+    const auto publicTerminal = static_cast<HwndTerminal*>(terminal);
+
+    auto lock = publicTerminal->_terminal->LockForWriting();
+    publicTerminal->_terminal->ClearSelection();
+
+    LOG_IF_WIN32_BOOL_FALSE(SetWindowPos(
+        publicTerminal->GetHwnd(),
+        nullptr,
+        0,
+        0,
+        static_cast<int>(width),
+        static_cast<int>(height),
+        0));
+
+    const SIZE windowSize{ static_cast<short>(width), static_cast<short>(height) };
+
+    RETURN_IF_FAILED(publicTerminal->_renderEngine->SetWindowSize(windowSize));
+
+    // Invalidate everything
+    publicTerminal->_renderer->TriggerRedrawAll();
+
+    // Convert our new dimensions to characters
+    const auto viewInPixels = Viewport::FromDimensions({ 0, 0 },
+                                                       { gsl::narrow<short>(windowSize.cx), gsl::narrow<short>(windowSize.cy) });

```

Comment:
It's weird to be re-casting the `short` out of the `long` here with `gsl:narrow`, both because we already had it in `short` form at one point and could have had it in a local (or in the method signature per my other comment) and because it doesn't make sense to use the throwable version of `gsl::narrow` when we're already certain that it had to already fit (so `gsl::narrow_cast` would be more appropriate.... but having the method signature `short` and filling those in here would be better yet.)